leetcode/iter_2.rs

```diff
@@ -0,0 +1,39 @@
+//Count Matches
+impl Solution {
+    pub fn number_of_matches(mut n: i32) -> i32 {
+        let mut matches = vec![];
+
+        while n > 1 {
+            matches.push(n / 2);
+
+            if n % 2 == 0 {
+                n /= 2;
+            } else {
+                n /= 2;
+                n += 1;
+            }
+        }
+
+        return matches.iter().fold(0, |sum, x| sum + x);
+    }
+}
+
+//Minimum Number of Moves
+impl Solution {
+    pub fn min_moves_to_seat(mut seats: Vec<i32>, mut students: Vec<i32>) -> i32 {
+        let mut moves = 0;
+
+        seats.sort();
+        students.sort();
+
+        for (i, j) in seats.iter().zip(students.iter()) {
+            if i > j {
```

Comment:
This can be simplified by using [`abs`](https://doc.rust-lang.org/std/primitive.i32.html#method.abs). (In fact, [`abs_diff`](https://doc.rust-lang.org/std/primitive.i32.html#method.abs_diff) introduced in Rust 1.60 is a better fit but unfortunately the Rust compiler on LeetCode is too old.)